Can't implement `Stringer` interface

Method `String` is written twice, default impl, and custom:

```rs
import "go:fmt"

fn main() {
  let a = A { a: "text" }
  fmt.Println(a.String())
}

struct A { a: string }

impl A {
  fn String(self) -> string {
    self.a + "asdf"
  }
}
```

Generated:

<details><summary>Details</summary>
<p>

```go
package main

import "fmt"

func main() {
	a := A{a: "text"}
	fmt.Println(a.String())
}

type A struct {
	a string
}

func (a A) String() string {
	return fmt.Sprintf("A { a: %v }", a.a)
}

func (a A) String() string {
	return a.a + "asdf"
}
```

</p>
</details> 

running:

```bash
# test
./main.go:18:12: method A.String already declared at ./main.go:14:12
```

When `.String()` doesn't called explicitly, declaration of `String()` is not used at all

--- a/crates/emit/src/definitions/structs.rs
+++ b/crates/emit/src/definitions/structs.rs
@@ -1,4 +1,5 @@
 use crate::Emitter;
+use crate::definitions::enum_layout::{ENUM_GO_STRINGER_METHOD, ENUM_STRINGER_METHOD};
 use crate::definitions::tags::{format_tag_string, interpret_field_attributes};
 use crate::names::generics::receiver_generics_string;
 use crate::names::go_name;
@@ -224,11 +225,12 @@ impl Emitter<'_> {
     }
 
     /// Returns the name for the auto-generated stringer method, or `None` if no
-    /// auto-generated method should be emitted.
+    /// auto-generated method should be emitted. Both Lisette casing (`string`,
+    /// `goString`) and Go casing (`String`, `GoString`) count as user-defined.
     ///
-    /// - No user `string` → auto-generated method is `"String"`
-    /// - User `string` only → auto-generated method is `"GoString"`
-    /// - User `string` + `goString` → `None` (skip, user covers both)
+    /// - No user stringer → auto-generated method is `"String"`
+    /// - User stringer only → auto-generated method is `"GoString"`
+    /// - User stringer + go-stringer → `None` (skip, user covers both)
     pub(super) fn stringer_method_name(&self, name: &str) -> Option<&'static str> {
         let qualified = format!("{}.{}", self.current_module, name);
         let methods = self
@@ -242,10 +244,14 @@ impl Emitter<'_> {
                 | Definition::TypeAlias { methods, .. } => Some(methods),
                 _ => None,
             });
-        match methods {
-            Some(m) if m.contains_key("string") && m.contains_key("goString") => None,
-            Some(m) if m.contains_key("string") => Some("GoString"),
-            _ => Some("String"),
+        let has_stringer = methods
+            .is_some_and(|m| m.contains_key("string") || m.contains_key(ENUM_STRINGER_METHOD));
+        let has_go_stringer = methods
+            .is_some_and(|m| m.contains_key("goString") || m.contains_key(ENUM_GO_STRINGER_METHOD));
+        match (has_stringer, has_go_stringer) {
+            (true, true) => None,
+            (true, false) => Some(ENUM_GO_STRINGER_METHOD),
+            _ => Some(ENUM_STRINGER_METHOD),
         }
     }
 }

--- a/tests/spec/emit/types.rs
+++ b/tests/spec/emit/types.rs
@@ -2631,6 +2631,89 @@ fn main() {
 }
 
 #[test]
+fn struct_user_pascal_string_method_suppresses_auto_stringer() {
+    let input = r#"
+struct A { a: string }
+
+impl A {
+  fn String(self) -> string {
+    self.a + "asdf"
+  }
+}
+
+fn main() {
+  let a = A { a: "text" }
+  let _ = a
+}
+"#;
+    assert_emit_snapshot!(input);
+}
+
+#[test]
+fn tuple_struct_user_pascal_string_method_suppresses_auto_stringer() {
+    let input = r#"
+struct UserId(int)
+
+impl UserId {
+  fn String(self) -> string {
+    "custom"
+  }
+}
+
+fn main() {
+  let u = UserId(1)
+  let _ = u
+}
+"#;
+    assert_emit_snapshot!(input);
+}
+
+#[test]
+fn enum_user_pascal_string_method_suppresses_auto_stringer() {
+    let input = r#"
+enum Color {
+  Red,
+  Blue,
+}
+
+impl Color {
+  fn String(self) -> string {
+    "custom"
+  }
+}
+
+fn main() {
+  let c = Color.Red
+  let _ = c
+}
+"#;
+    assert_emit_snapshot!(input);
+}
+
+#[test]
+fn struct_user_pascal_string_and_go_string_methods() {
+    let input = r#"
+struct Point { x: int, y: int }
+
+impl Point {
+  fn String(self) -> string {
+    "custom display"
+  }
+
+  fn GoString(self) -> string {
+    "custom debug"
+  }
+}
+
+fn main() {
+  let p = Point { x: 1, y: 2 }
+  let _ = p
+}
+"#;
+    assert_emit_snapshot!(input);
+}
+
+#[test]
 fn impl_methods_share_local_variable_name() {
     let input = r#"
 pub struct Calc {}

